lessons/advanced/gen-stage.md

```diff
@@ -142,9 +142,9 @@ defmodule GenstageExample.ProducerConsumer  do
 end
 ```
 
-You may have noticed with our producer-consumer we've introduced a new function `handle_events/3` and option in `init/1`.  With the `subscribe_to` option we instruct GenStage to put us into communcation with a specific producer.
+You may have noticed with our producer-consumer we've introduced a new option in `init/1` and a new function: `handle_events/3`.  With the `subscribe_to` option, we instruct GenStage to put us into communcation with a specific producer.
 
-The `handle_events/3` method is our workhorse, where we receive our incoming events, process them, and return our transformed set.  As we'll see consumers are implemented in much the same way but the important difference is what our `handle_events/3` method returns and how it's used.   When we label our process a process-consumer the second argument of our tuple, `numbers` in our case, is used to meet the demand of consumers downstream.  In consumers this value is discarded.
+The `handle_events/3` method is our workhorse, where we receive our incoming events, process them, and return our transformed set.  As we'll see consumers are implemented in much the same way, but the important difference is what our `handle_events/3` method returns and how it's used.   When we label our process a process-consumer, the second argument of our tuple — `numbers` in our case — is used to meet the demand of consumers downstream.  In consumers this value is discarded.
```

Comment:
I dont know the rules of english well enough to tell if this is proper hyphenization. Not questioning, just would like someone else with a better grasp to sign off on it.